shell/platform/linux/fl_engine_impl_egl.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_ENGINE_IMPL_EGL_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_ENGINE_IMPL_EGL_H_
+
+#include <EGL/egl.h>
+
+#include "flutter/shell/platform/linux/fl_engine_impl.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_DERIVABLE_TYPE(FlEngineImplEGL,
```

Comment:
Orthogonally to the way graphics variations are broken down: I think this is close to the right structure, but there's still a conceptually mismatch between this engine object and what the engine is on other platform. The engine should be a graphics-details-agnostic object which deals with running the engine, but doesn't know the details of how it's displayed (and in fact should support being run in a headless mode with no drawing at all); the Flutter view manages that part. So the EGL/X11/Wayland objects shouldn't be `FlEngine*`, but something like `FlView*`. The engine should have a concept of being attached (or not) to a view, and forwarding drawing-related things to that view.

That's what I was trying to describe in my comment in the previous review round, but I think I ended up focusing too much on the public vs private API aspect rather than the conceptual structure.

See [here](https://github.com/flutter/engine/blob/master/shell/platform/darwin/macos/framework/Source/FlutterEngine.mm#L314) for instance for the macOS version of this. (Although I see that someone changed it from being the FlutterView to being an open gl context at some point after I wrote it, so maybe use an interface object separate from FlView to give us more flexibility.)